generate_new_lesions/augment_lesions_intra_dataset.py

```diff
@@ -0,0 +1,495 @@
+
+Activate SCT conda environment:
+    source ${SCT_DIR}/python/etc/profile.d/conda.sh
+    conda activate venv_sct
+
+Run:
+    python augment_lesions_intra_dataset.py
+
+NOTE: BIDS structure is expected for the input data.
+"""
+import os
+import sys
+import time
+import yaml
+import argparse
+import numpy as np
+import pandas as pd
+from scipy.ndimage import binary_dilation, generate_binary_structure
+from sklearn.model_selection import train_test_split
+
+from spinalcordtoolbox.image import Image, zeros_like
+from spinalcordtoolbox.resampling import resample_nib
+
+from utils import get_centerline, get_lesion_volume, fetch_subject_and_session, \
+    generate_histogram, extract_lesions
+
+def get_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-path-data", type=str, required=True,
+                        help="Path to BIDS dataset containing both healthy controls and patients with lesions "
+                             "(e.g., basel-mp2rage)")
+    parser.add_argument("-path-out", type=str, required=True,
+                        help="Path where the augmented dataset will be saved.")
+    parser.add_argument("-num", default=100, type=int, help="Total number of newly generated subjects. Default: 100")
+    parser.add_argument("-seed", default=99, type=int, help="Random seed used for subject mixing. Default: 99")
+    parser.add_argument("-resample", default=False, action='store_true',
+                        help="Resample the augmented images to the resolution of pathological dataset. Default: False")
+    parser.add_argument("-use-n-healthy", default=1, type=int,
+                        help="Number of healthy controls to use for augmentation Default: num_patho // 2")
+    parser.add_argument('--fraction_patho_data', default=1.0, type=float,
+                    help='Fraction of pathology data to be used for training. '
+                    'Use 0.5 for half, 0.25 for 1/4th, etc. ') 
+    parser.add_argument('-split', nargs='+', required=False, type=float, default=[0.6, 0.2, 0.2],
+                        help='Ratios of training, validation and test splits lying between 0-1. Example: --split 0.6 0.2 0.2')
+    parser.add_argument("-qc", default=False, action='store_true', 
+                        help="Perform QC using sct_qc. QC will be saved under -path-out. Default: False")
+    # parser.add_argument("-histogram", default=False, action='store_true', help="Create histograms. Default: False")
+    # parser.add_argument("-min-lesion-vol", "--min-lesion-volume", default=200, type=float,
+    #                     help="Minimum lesion volume in mm^3. Default: 200")
+
+    return parser
+
+
+def insert_lesion(im_augmented, im_augmented_lesion, im_patho_data, im_patho_sc_dil_data, patho_lesion_data, 
+                  im_healthy_sc_data, coords, new_position, lesion_sc_ratio_patho):
+    """"
+    Insert lesion from the bounding box to the im_augmented
```

Comment:
add doctrings explaining each input variable

Reply:
suggestions for variable:
- im_augmented --> im_target
- im_augmented_lesion --> mask_lesion_target
- im_patho_data --> im_source
- patho_lesion_data --> mask_lesion_source
- im_healthy_sc_data --> mask_sc_target
- coords --> remove from input and get it from an API inside this function
- new_position --> 
- lesion_sc_ratio_patho --> not relevant according to discussion with @naga-karthik 

i suggest everywhere in the code to use the terms source and target.

Reply:
Thank you for the suggestions! Refactoring done in commits https://github.com/ivadomed/model_seg_sci/pull/45/commits/e2ac381f9ed44ecdad0b696273186082d920b604 and https://github.com/ivadomed/model_seg_sci/pull/45/commits/b3034ecde65a6f8d30998749b5c85f72a6626c84